tools/contrib/junit_xml/__init__.py

```diff
@@ -0,0 +1,322 @@
+#!/usr/bin/env python
```

Comment:
It cannot be done in the manner of run_tests.py and achieve the goals. 
Run_tests.py pays the cost of a subshell and secondary python process. This code is intended to catch meaningful exceptions so that failures can be accurately represented in Jenkins without reading many megabytes of logs.  activating a virtualenv within a python process is somewhat black magic, but installing modules via pip inside a python process and then using them will require buckets of it. 
There is no intention to add any modules beyond junit_xml in the future, and it may be worth it at some point to generate the xml myself. 

As a further point if this design is a real problem, someone could have replied when I explicitly raised the question as a concern on slack.